apps/passport-server/test/frogcrypto.spec.ts

```diff
@@ -80,27 +72,63 @@ describe("frogcrypto functionality", function () {
     expect(feed.active).to.be.true;
     expect(feed.private).to.be.false;
 
-    const response = await getFrog(feed);
-    expect(response.success).to.be.true;
+    await testGetFrog(feed, DATE_EPOCH_1H);
+  });
 
-    expect(response.value?.actions.length).to.eq(1);
-    const populateAction = response.value?.actions[0] as AppendToFolderAction;
-    expect(populateAction.type).to.eq(PCDActionType.AppendToFolder);
-    expect(populateAction.folder).to.eq("FrogCrypto");
-    expect(populateAction.pcds[0].type).to.eq(EdDSAFrogPCDPackage.name);
+  it("should be able to get frog even if feed is private", async () => {
+    const feed = FROGCRYPTO_FEEDS[1];
+    expect(feed.active).to.be.true;
+    expect(feed.private).to.be.true;
 
-    frogPCD = await EdDSAFrogPCDPackage.deserialize(populateAction.pcds[0].pcd);
-    expect(frogPCD.claim.data.ownerSemaphoreId).to.eq(
-      identity.commitment.toString()
+    await testGetFrog(feed, DATE_EPOCH_1H);
+  });
+
+  it("should not be able to get frog if feed is inactive", async () => {
+    const feed = FROGCRYPTO_FEEDS[2];
+    expect(feed.active).to.be.false;
+    expect(feed.private).to.be.true;
+
+    await testGetFrogFail(feed, DATE_EPOCH_1H, "not active");
+  });
+
+  it("should be able to get frog if after cooldown", async () => {
+    const feed = FROGCRYPTO_FEEDS[0];
+    expect(feed.active).to.be.true;
+    expect(feed.private).to.be.false;
+    expect(feed.cooldown).to.eq(60);
+
+    await testGetFrog(feed, DATE_EPOCH_1H);
+    await testGetFrog(feed, DATE_EPOCH_1H1M);
+  });
+
+  it("should not be able to get frog if before cooldown", async () => {
+    const feed = FROGCRYPTO_FEEDS[0];
+    expect(feed.active).to.be.true;
+    expect(feed.private).to.be.false;
+    expect(feed.cooldown).to.eq(60);
+
+    await testGetFrog(feed, DATE_EPOCH_1H1M);
+    await testGetFrogFail(
+      feed,
+      DATE_EPOCH_1H1M59S,
+      "Next fetch available at 3720000"
     );
   });
 
-  it("should be able to get frog even if feed is private", async () => {
-    const feed = FROGCRYPTO_FEEDS[1];
+  it("should get 404 if no frogs are available", async () => {
+    const feed = FROGCRYPTO_FEEDS[3];
     expect(feed.active).to.be.true;
     expect(feed.private).to.be.true;
+    expect(feed.biomes).to.have.same.members([Biome.TheWrithingVoid]);
+
+    await testGetFrogFail(feed, DATE_EPOCH_1H, "Frog Not Found");
+  });
```

Comment:
idk if 404 is the right thing, rather than a 500 here. why are no frogs available here?

Reply:
Technically this should be a 5xx error where we misconfigured the feed. Realistically we never expect this to happen. `404 Frog Not Found` is just a funny error :D